src/decoder.rs

```diff
@@ -855,6 +856,11 @@ impl<'a> Compiler<'a> {
                 let da = Box::new(self.compile_format(a, next)?);
                 Ok(Decoder::RepeatUntilLast(expr.clone(), da))
             }
+            Format::ForEach(expr, lbl, a) => {
+                // FIXME probably not right
```

Comment:
Perhaps this comment (and the others like it) should clarify that it's the `next` value we're passing in that is probably not right 🤔

Reply:
Indeed. I think this one might even be correct enough we don't need the comment.